enzy_htp/analysis/cavity.py

```diff
@@ -0,0 +1,156 @@
+from enzy_htp import config as eh_config
+from enzy_htp.core import file_system as fs
+
+sp = PDBParser()
+CAVITY_IDENTIFICATION_METHODS: Dict[str, Callable[..., List[Cavity]]] = {
+    "mole2": interface.mole2.identify_cavities
+}
+
+def identify_stru_cavities(stru: Structure,
+        work_dir: str = None,
+        engine: str = "mole2",
+        **kwargs
+    ) -> List[Cavity]:
+    """Identifies cavities in a Structure instance. 
+    Currently available engine:
+    * mole2;
+
+    Args:
+        stru (Structure): The structure instance to detect cavities from.
+        work_dir (str, optional): Directory to do work in. Defaults to system.SCATCH_DIR if not supplied.
+        engine (str): The engine to use for cavity identification. Defaults to "mole2" (The only available one at present).
+        **kwargs: Engine-specific parameters in keyword arguments. 
+
+    Returns:
+        cavities (List[Cavity]): A list of Cavity objects.
+
+    Details:
+        * mole2 specific arguments:
+            - non_active_residues (List[Residue], optional): Residues that should be skipped.
+            - probe (float, optional): Probe radius to use in A. Defaults to Mole2Config.PROBE if not supplied.
+            - inner (float, optional): Inner radius to use in A. Defaults to Mole2Config.INNER if not supplied.
+            - mesh_density (float, optional): Mesh density to use in A. Defaults to Mole2Config.MESH_DENSITY if not supplied.
+            - ignore_hetatm (bool, optional): TODO (CJ)
+            - use_mono (bool, optional): Indicate if mono need to be used during run time. Defaults to true.
+
+    """
+    if work_dir is None:
+        work_dir = eh_config['system.SCRATCH_DIR']
+
+    cavities = CAVITY_IDENTIFICATION_METHODS[engine](stru=stru, work_dir=work_dir, **kwargs)
+    
+    return cavities
+
+def _choose_cavity(cavity_list: List[Cavity], 
+        composing_residues: List[Residue] = None, 
+        contain_ligand: str = None, 
+        target_cavity: Cavity = None
+    ) -> Tuple[Cavity, int]:
+    """
+    Selects the most similar cavity to the focus cavity based on residue key overlap.
+    
+    Args:
+        cavity_list: List of Cavity objects to compare. All cavities should be in the same Structure instance.
+        composing_residues (List[Residue], optional): A list of Residue instances composing the cavity. Mutually exclusive with `contain_ligand` or `target_cavity`.
+        contain_ligand (str, optional): The PyMOL-formatted pattern selecting the ligand in the cavity. Mutually exclusive with `target_cavity` or `composing_residues`.
+        target_cavity (Cavity, optional): The target cavity to track throughout the ensemble (by most overlapped residues). Mutually exclusive with `composing_residues` or `contain_ligand`.
+    
+    Returns:
+        result (Tuple[Cavity, int]): 
+            * Cavity: The Cavity object from cavity_list with highest residue key overlap with focus_cavity
+            * int: max_overlap value.
```

Comment:
Make it a number from 0 to 1 so that it is unified regradless of ligand/residue_list supplied. For the ligand case, it can be (the number of atoms contained by the final cavity)/(the total number of atoms contained by all cavities)